### DISCUSSION TITLE
Change some struct/value types to class/reference type

### DISCUSSION THREAD

**Discussion by cmoha** [BriefFiniteElementNet/BriefFiniteElement.Net#110] [Ideas]
Dear Eng Ehsan,
Is it possible to change the type of some data types such as `LoadType, Settlement, etc..` to be reference type.
This will be very handy in WPF applications and will make databinding works.

**Comment by epsi1on** [comment]
Hi,
Did used struct for those types only because they are sort of primitive types (like int, double, etc.).
Most common structures are `Force`, `Displacement`, `BendingStressTensor`, `CauchyStressTensor`, `Constraint`, `LoadCase`, `Settlement`, `StrainTensor`, `Vector`, `Point` and `IsoPoint`. Which ones do you suggest to convert into class?

**Reply by cmoha** [reply]
I suggest `Force, Displacement, Constraint, LoadCase and Settlement.`
Because These types are more likely to be in an interaction with a gui.

**Reply by epsi1on** [reply]
Let me think about it, not sure it meet the conditions of class instead of struct

**Reply by epsi1on** [reply]
also take a look at here:
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/choosing-between-class-and-struct

![image](https://user-images.githubusercontent.com/7839141/106557790-cd7c1580-6537-11eb-8417-105d0cc24fcd.png)

Except for second condition, all other conditions are met

**Comment by epsi1on** [comment]
Also about using with WPF, i think this library better to be not used directly with WPF. I mean ideally it is better to create the a data model for your project yourself, and that model only use this library for calculations etc. For example a building which have elevations, grids etc can be custom data model, then that model creates a Finite Element model as instance of `Model` class with a method like `CustomBuilding.GenerateFEModel()`

**Reply by cmoha** [reply]
I understand your point. But I want to use these types directly instead of creating and copying classes then at linking time, there will a conversion between the project and bfe library.

**Comment by epsi1on** [comment]
inactivity
